notebooks/gnn/mag240m_pg.ipynb

```diff
@@ -0,0 +1,405 @@
+   "metadata": {},
+   "source": [
+    "## Setup variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# size of dataset\n",
+    "class Dataset:\n",
+    "    def __init__(self):\n",
+    "        self.num_papers  = 121751666\n",
+    "        self.num_authors = 122383112\n",
+    "        self.num_institutions = 25721\n",
+    "        self.num_paper_features = 768\n",
+    "        self.num_classes = 153\n",
+    "        self.num_cite_edges = 1297748926\n",
+    "        self.num_write_edges = 386022720\n",
+    "        self.num_affiliated_edges = 44592586\n",
+    "    \n",
+    "    def adjust_by(self, factor):\n",
+    "        self.num_papers = math.floor(self.num_papers * factor)\n",
+    "        self.num_authors = math.floor(self.num_authors * factor)\n",
+    "        self.num_institutions = math.floor(self.num_institutions * factor)\n",
+    "        self.num_paper_features = math.floor(self.num_paper_features * factor)\n",
+    "        self.num_classes = math.floor(self.num_classes * factor)\n",
+    "        self.num_cite_edges = math.floor(self.num_cite_edges * factor)\n",
+    "        self.num_write_edges = math.floor(self.num_write_edges * factor)\n",
+    "        self.num_affiliated_edges = math.floor(self.num_affiliated_edges * factor)        \n",
+    "\n",
+    "dataset = Dataset()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Directories and Files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mag_dir = \"/home/brad/Notebooks/dataset/mag240m_kddcup2021/processed\"\n",
+    "\n",
+    "# Features\n",
+    "paper_feature_file = mag_dir + \"/paper/node_feat.npy\"\n",
+    "paper_label_file = mag_dir + \"/paper/node_label.npy\"\n",
+    "paper_year_file = mag_dir + \"/paper/node_year.npy\"\n",
+    "\n",
+    "# Edges\n",
+    "auth_institute_file = mag_dir + \"/author___affiliated_with___institution/edge_index.npy\"\n",
+    "auth_write_file = mag_dir + \"/author___writes___paper/edge_index.npy\"\n",
+    "auth_cites_file = mag_dir + \"/paper___cites___paper/edge_index.npy\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#!nvidia-smi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Specify the GPUs to use\n",
+    "import os\n",
+    "os.environ[\"CUDA_VISIBLE_DEVICES\"]=\"0\"\n",
+    "\n",
+    "num_gpus = 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# it takes 16 GPUs to fully load the data\n",
+    "# compute the percent of data to be loaded\n",
+    "# This will be used later\n",
+    "percent_data_factor = num_gpus / 200"
```

Comment:
Can you comment on why 200 was chosen, since this seems like useful knowledge.

Reply:
oops - that was a testing number that slipped in